lms/djangoapps/shoppingcart/views.py

```diff
@@ -691,6 +691,16 @@ def postpay_callback(request):
         # See if this payment occurred as part of the verification flow process
         # If so, send the user back into the flow so they have the option
         # to continue with verification.
+
+        # Only orders where order_items.count() == 1 might be attempting to upgrade
+        attempting_upgrade = request.session.get('attempting_upgrade', False)
+        if attempting_upgrade and result['order'].has_items(CertificateItem):
+            course_id = result['order'].orderitem_set.all().select_subclasses()[0].course_id
+            course_enrollment = CourseEnrollment.get_enrollment(request.user, course_id)
+            if course_enrollment:
+                course_enrollment.emit_event(EVENT_NAME_USER_UPGRADED)
+            request.session['attempting_upgrade'] = False
```

Comment:
Elsewhere, we seem to be calling `session.save()`.  Do we need to do this here too?  (And on line 711)